includes/class-wc-payments-payment-request-button-handler.php

```diff
@@ -262,22 +274,38 @@ public function get_product_data() {
 			}
 		}
 
-		$data          = [];
-		$items         = [];
-		$product_price = $this->get_product_price( $product );
+		try {
+			$price = $this->get_product_price( $product );
+		} catch ( Invalid_Price_Exception $e ) {
+			Logger::log( $e->getMessage() );
+			return false;
+		}
+
+		$data  = [];
+		$items = [];
 
 		$items[] = [
 			'label'  => $product->get_name(),
-			'amount' => WC_Payments_Utils::prepare_amount( $product_price, $currency ),
+			'amount' => WC_Payments_Utils::prepare_amount( $price, $currency ),
 		];
 
-		$tax = $this->prices_exclude_tax() ? wc_format_decimal( wc_get_price_including_tax( $product ) - $product_price ) : 0;
-		if ( wc_tax_enabled() ) {
-			$items[] = [
-				'label'   => __( 'Tax', 'woocommerce-payments' ),
-				'amount'  => WC_Payments_Utils::prepare_amount( $tax, $currency ),
-				'pending' => ( 0 === $tax ? true : false ),
-			];
+		$total_tax = 0;
+		if ( wc_tax_enabled() && $this->prices_exclude_tax() ) { // Only enter when taxes are enabled, but not included.
+			// Follows the way `WC_Cart_Totals::get_item_tax_rates()` works.
+			$tax_class = $product->get_tax_class();
+			$rates     = WC_Tax::get_rates( $tax_class );
+			// No cart item, `woocommerce_cart_totals_get_item_tax_rates` can't be applied here.
+
+			// Normally there should be a single tax, but `calc_tax` returns an array, let's use it.
+			foreach ( WC_Tax::calc_tax( $price, $rates, false ) as $tax ) {
+				$total_tax += $tax;
+
+				$items[] = [
+					'label'   => __( 'Tax', 'woocommerce-payments' ),
+					'amount'  => WC_Payments_Utils::prepare_amount( $tax, $currency ),
+					'pending' => ( 0 === $tax ? true : false ),
```

Comment:
Nitpick. This ternary operator can be replaced with` 0 === $tax`

Reply:
Fixed in https://github.com/Automattic/woocommerce-payments/pull/4649/commits/21a37b3caf5c43e5feb66496892ab589170eb407